cmd/bonanza_browser/pretty.go

```diff
@@ -0,0 +1,321 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"path"
+	"slices"
+	"time"
+
+	model_core "bonanza.build/pkg/model/core"
+	browser_pb "bonanza.build/pkg/proto/browser"
+	model_core_pb "bonanza.build/pkg/proto/model/core"
+	model_filesystem_pb "bonanza.build/pkg/proto/model/filesystem"
+	"bonanza.build/pkg/storage/object"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+	g "maragu.dev/gomponents"
+	h "maragu.dev/gomponents/html"
+)
+
+// messagePrettyRenderer renders the decoded payload of an object
+// as a pretty-printed object intended for humans. It assumes the
+// contents are a Protobuf message that can be converted to JSON.
+type messagePrettyRenderer struct{}
```

Comment:
Instead of adding this type, could we change `messageJSONPayloadRenderer` and `messageListJSONPayloadRenderer` to have a `customRenderer` field? Then we can make the pretty printing work both for plain message objects and message list objects.

Reply:
@EdSchouten I'll do that, but we'll still need the `messagePrettyRenderer` since it needs to have its own implementation of `queryParameter` and `name`.

Reply:
Done 👍